### ISSUE TITLE
Is there any plan to release code for Classification

### ISSUE THREAD

**Issue by NickYangMin** [jshilong/FisherPruning#20]
Is there any plan to release code for Classification? Or where else can I find the code.

**Comment by yumath** [comment]
Similar question. Can the existing FisherPruning applied to detection be directly applied to classification?

**Comment by jshilong** [comment]
Actually, this hook can be directly applied to `mmclassification` except for architectures with depth/group-conv

**Comment by Janspiry** [comment]
@jshilong， 
Hi，I used the ResNet50 for classification task in the ImageNet benchmark.
 The baseline Top-1 is **76.15%**, I got the Top-1 **75.61%** in Flops-50% model and **72.50%** in Flops-25% model, respectively (Flops-x represents the remaining Flops). 
The configures is the same with given paper, which using the pruning lr=0.004, finetune lr =0.1 and batch_size=256 in mmclassification style.
![image](https://user-images.githubusercontent.com/32922494/213910252-0481de5a-9bab-4991-8765-4fee072ab63b.png). 

This is somewhat different from the paper report, especially in the case of Flops-25%. Is there any possible reason for this?

**Comment by jshilong** [comment]
This code is refactored into a hook  for the openmmlab series algorithm.
When we were doing this project, the mmclassification was still very
primitive version（Many algorithms in the paper were not supported at that
time）which could not meet our experimental requirements. We used a project
called xxxxtype  inside sensetime. The baseline of res50 is 76.79 as we
mentioned in table1, which may have a lot of gaps with mmcls setting. I
think your result can also be used in your paper as long as you clearly
state your baseline because the implementation of this hook is consistent
with our paper.



 Liangwei Jiang ***@***.***>于2023年1月22日 周日18:10写道：

**Comment by Janspiry** [comment]
Happy New Year and thank you very much for your patient and quick reply.

I read the supplemental material carefully and confirmed that the previous training configuration is aligned with the original configuration, e.g. step=25.

I think the general results are usable. For the Flops-25 result, does this have anything to do with finetuning learning rate? (as I achieved better results with bs=256 and lr=0.05)
I would appreciate if you have experiments with relevant parameters at low remain Flops.
